### PR TITLE
Refactors Sign Language & Fixes Inconsistency

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Refactors Sign Language code so instead of copy-pasting the same giant wall of checks we can just use a proc.
Also now checks to see if your limb is disabled, which fixes people with disabled robotic limbs being able to sign still.
Finally, the tongue only has ORGAN_UNREMOVABLE if you attained it from the trait. I've been told that the tongue could be attained from meateors and I think that's funny as hell so I swapped that over.

## Why It's Good For The Game
Cleaner code makes it easier to tell what's going on. Also makes it easier for other people to do their own checks for open hands in the future, if they needed to.

## Changelog

:cl: Wallem
fix: People with robotic limbs are now incapable of using Sign Language if their limbs are disabled.
refactor: Refactored Sign Language
/:cl:

### CODE REVIEW

File: code/datums/quirks/neutral.dm [hunk 1]
```diff
@@ -313,6 +313,8 @@
 
 	var/obj/item/organ/tongue/tied/new_tongue = new(get_turf(human_holder))
 	new_tongue.Insert(human_holder)
+	// Only tongues of people with this quirk can't be removed. Manually spawned or found tongues can be.
+	new_tongue.organ_flags = ORGAN_UNREMOVABLE
```
Comment (MrMelbert) [inline code/datums/quirks/neutral.dm:317]:
```suggestion
	new_tongue.organ_flags |= ORGAN_UNREMOVABLE
```
you never know...

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -1318,3 +1318,21 @@
 	to_chat(src, span_danger("You shove [target.name] into [name]!"))
 	log_combat(src, target, "shoved", "into [name]")
 	return COMSIG_CARBON_SHOVE_HANDLED
+
+
+// Checks to see how many hands this person has to sign with.
+/mob/living/carbon/proc/check_signables_state()
+	var/mob/living/carbon/mute = src
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/carbon.dm:1325]: why do you cast src to a carbon when it's already a carbon?

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -1318,3 +1318,21 @@
...
+	var/obj/item/bodypart/left_arm = mute.get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/right_arm = mute.get_bodypart(BODY_ZONE_R_ARM)
+	var/empty_indexes = get_empty_held_indexes()
+	if(length(empty_indexes) == 0 || (length(empty_indexes) < 2 && ((!left_arm || left_arm.bodypart_disabled) || (!right_arm || right_arm.bodypart_disabled))))//All existing hands full, can't sign
+		return 1 // These aren't booleans
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/carbon.dm:1330]: You should make these into defines if they're not booleans
like `define VALID_SIGNER 0`, `define FULL_HANDS 1`, `define NO_ARMS 2`, etc

Comment (Wallemations) [reply code/modules/mob/living/carbon/carbon.dm:1330]: Would I not have to define those for the other files too? Or do you mean for me to add those to the actual _DEFINES/mobs.dm file?

File: code/modules/mob/living/carbon/carbon.dm (hunk 2 above)
Comment (Wallemations) [reply code/modules/mob/living/carbon/carbon.dm:1325]: overlooked that, I'll fix

File: code/modules/mob/living/carbon/carbon.dm (hunk 3 above)
Comment (Wallemations) [reply code/modules/mob/living/carbon/carbon.dm]: Hope that was what you meant, I swapped it

File: code/modules/mob/living/carbon/carbon.dm [hunk 4]
```diff
@@ -1318,3 +1318,20 @@
 	to_chat(src, span_danger("You shove [target.name] into [name]!"))
 	log_combat(src, target, "shoved", "into [name]")
 	return COMSIG_CARBON_SHOVE_HANDLED
+
+
+// Checks to see how many hands this person has to sign with.
+/mob/living/carbon/proc/check_signables_state()
+	var/obj/item/bodypart/left_arm = src.get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/right_arm = src.get_bodypart(BODY_ZONE_R_ARM)
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/carbon.dm:1326]:
```suggestion
	var/obj/item/bodypart/left_arm = get_bodypart(BODY_ZONE_L_ARM)
	var/obj/item/bodypart/right_arm = get_bodypart(BODY_ZONE_R_ARM)
```

File: code/modules/mob/living/carbon/carbon.dm (hunk 3 above)
Comment (MrMelbert) [reply code/modules/mob/living/carbon/carbon.dm]: LGTM

File: code/modules/mob/living/carbon/carbon.dm (hunk 4 above)
Comment (Wallemations) [reply code/modules/mob/living/carbon/carbon.dm]: Right, I'll fix that & also the handcuffed check too

File: code/modules/mob/living/carbon/carbon.dm [hunk 5]
```diff
@@ -1318,3 +1318,20 @@
...
+	if(length(empty_indexes) == 0 || (length(empty_indexes) < 2 && ((!left_arm || left_arm.bodypart_disabled) || (!right_arm || right_arm.bodypart_disabled))))//All existing hands full, can't sign
+		return SIGN_HANDS_FULL // These aren't booleans
+	if((!left_arm || left_arm.bodypart_disabled) && (!right_arm || right_arm.bodypart_disabled))//Can't sign with no arms!
+		return SIGN_ARMLESS
+	if(HAS_TRAIT(src, TRAIT_HANDS_BLOCKED) || HAS_TRAIT(src, TRAIT_EMOTEMUTE))
+		return SIGN_TRAIT_BLOCKED
+	if(handcuffed) // Cuffed, usually will show visual effort to sign
+		return SIGN_CUFFED
+	if(length(empty_indexes) == 1 || (!left_arm || left_arm.bodypart_disabled) || (!right_arm || right_arm.bodypart_disabled)) // One arm gone
+		return SIGN_ONE_HAND
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/carbon.dm:1337]: You duplicate several times in the if statements, specifically (!left_arm || left_arm.bodypart_disabled) and (!right_arm || right_arm.bodypart_disabled) which could be held as variables instead and compared that way, making the if statements more readable.

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon.dm (hunk 5 above)
Comment (Wallemations) [reply code/modules/mob/living/carbon/carbon.dm:1337]: I never realized that I could use variables for or statements, I'll give it a shot

**Pull request merged**